worker/fanconfigurer/fanconfigurer.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package fanconfigurer
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/clock"
+
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/utils/scriptrunner"
+	"github.com/juju/juju/watcher"
+	"github.com/juju/juju/worker/catacomb"
+)
+
+var logger = loggo.GetLogger("juju.worker.fanconfigurer")
+
+type FanConfigurer struct {
+	catacomb catacomb.Catacomb
+	config   FanConfigurerConfig
+	mu       sync.Mutex
+	enabled  bool
+}
+
+type FanConfigurerFacade interface {
+	FanConfig() ([]config.FanConfigLine, error)
+	WatchForFanConfigChanges() (watcher.NotifyWatcher, error)
+}
+
+type FanConfigurerConfig struct {
+	Facade FanConfigurerFacade
+}
+
+// processNewConfig acts on a new fan config.
+func (fc *FanConfigurer) processNewConfig() error {
+	logger.Debugf("Processing new fan config")
+	fc.mu.Lock()
+	defer fc.mu.Unlock()
+
+	fanConfig, err := fc.config.Facade.FanConfig()
+	if err != nil {
+		return err
+	}
+	if len(fanConfig) == 0 {
+		logger.Debugf("Fan not enabled")
+		// TODO(wpk) 2017-08-05 We have to clean this up!
+		return nil
+	}
+
+	// TODO(wpk) 2017-08-09 Verify that it is indeed the best way to configure fan,
+	// also check all the results code
+	for i, fan := range fanConfig {
+		logger.Debugf("Adding config for %d: %s %s", i, fan.Underlay, fan.Overlay)
+		line := fmt.Sprintf("fanatic enable-fan -u %s -o %s", fan.Underlay, fan.Overlay)
+		result, err := scriptrunner.RunCommand(line, os.Environ(), clock.WallClock, 5000*time.Millisecond)
```

Comment:
The clock should be passed in to the worker.

Reply:
Done